index.md

```diff
@@ -3,87 +3,115 @@
+
+- [JuliaDiff/ChainRules](https://www.juliadiff.org/ChainRulesCore.jl/stable/): Ecosystem for backend-agnostic forward and reverse rules.
+  - [JuliaDiff/ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl): Core API for users to add rules to their package.
+  - [JuliaDiff/ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl/): Rules for Julia Base and standard libraries.
+  - [JuliaDiff/ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl/): Tools for testing rules defined with ChainRulesCore.jl.
+- [ThummeTo/ForwardDiffChainRules.jl](https://github.com/ThummeTo/ForwardDiffChainRules.jl): Translate rules from ChainRulesCore.jl to make them compatible with ForwardDiff.jl
+- [JuliaDiff/DiffRules.jl](https://github.com/JuliaDiff/DiffRules.jl): Scalar rules used by ForwardDiff.jl
+- [FluxML/ZygoteRules.jl](https://github.com/FluxML/ZygoteRules.jl): Some rules used by Zygote.jl (mostly deprecated in favor of ChainRules.jl).
 
 ### Interface
-- [AbstractDifferentiation.jl](https://github.com/JuliaDiff/AbstractDifferentiation.jl): AD backend-agnostic interface for algorithms that rely on derivatives, gradients, Jacobians, Hessians, etc.
-
-## Links for discussion and more information
-Discussions on JuliaDiff and its uses may be directed to the [Julia Discourse forum](https://discourse.julialang.org/)
-The [autodiff.org](http://www.autodiff.org/) site serves as a portal for the academic community, though it is often out-of-date.
-The ChainRules project maintains a [list of recommend reading/watching](https://www.juliadiff.org/ChainRulesCore.jl/stable/FAQ.html#Where-can-I-learn-more-about-AD-?) for those after more information.
-Finally, automatic differentiation techniques have been implemented in a variety of languages.
-If you would prefer not to use Julia, see the [wikipedia page](http://en.wikipedia.org/wiki/Automatic_differentiation) for a comprehensive list of available packages.
+
+- [AbstractDifferentiation.jl](https://github.com/JuliaDiff/AbstractDifferentiation.jl): Backend-agnostic interface for algorithms that rely on derivatives, gradients, Jacobians, Hessians, etc.
+
+### Exotic
+
+- [JuliaDiff/SparseDiffTools.jl](https://github.com/JuliaDiff/SparseDiffTools.jl): Exploit sparsity to speed up FiniteDiff.jl and ForwardDiff.jl, as well as other algorithms.
+- [gaurav-arya/StochasticAD.jl](https://github.com/gaurav-arya/StochasticAD.jl): Differentiation of functions with stochastic behavior (experimental)
+
+### Differentiating through more stuff
+
+Some complex algorithms are not natively differentiable, which is why derivatives have been implemented in the following packages:
+
+- [SciML](https://github.com/SciML): For a lot of different domains of scientific machine learning: differential equations, linear and nonlinear systems, optimization problems, etc.
+- [TuringLang/DistributionsAD.jl](https://github.com/TuringLang/DistributionsAD.jl): For probability distributions
```

Comment:
Not sure if this should be listed, it's a type piracy mess and the intention is to move definitions and workarounds to upstream packages such as Distributions, ReverseDiff etc.
```suggestion
```

Reply:
I would still mention it, if only to tell people exactly that